### PR TITLE
Add Assessment Registry using iframe

### PR DESCRIPTION
**Pull request by samshara**
- Add assessment registry
- Depends on https://github.com/the-deep/client/tree/feature/only-ary

### how to run assessment registry 
1. Deploy the branch https://github.com/the-deep/client/tree/feature/only-ary
2. Update REACT_APP_ASSESSMENT_REGISTRY_END env to the deployed endpoint

If you want to run assessment registry in local environment
1. Copy all the code from (https://github.com/the-deep/client/tree/feature/only-ary) to a client_old folder under [deeper](https://github.com/the-deep/deeper) root directory
2.  Update docker-compose.yml 
```
  client_old:
      image: thedeep/deep-client:latest-old
      build:
        context: ./client_old/
        cache_from:
          - thedeep/deep-client:latest
      tty: true
      env_file:
        - .env
      command: bash -c '/code/scripts/run_develop.sh'
      volumes:
        - ./client_old/:/code
      ports:
        - '3100:3000'
```
3. Update REACT_APP_ASSESSMENT_REGISTRY_END `REACT_APP_ASSESSMENT_REGISTRY_END=http://localhost:3100` 
4. run `docker-compose up`

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

## This PR contains valid:

- [ ] permission checks
- [ ] translations

### CODE REVIEW

File: app/Base/configs/routes.tsx [hunk 1]
```diff
@@ -328,6 +328,17 @@ const exportRoute = wrap({
...
 
+const assessmentEditRoute = wrap({
+    parent: { path: taggingRoute.path },
+    path: '/assessments/:leadId(\\d+)/',
+    title: 'Assessment Edit',
+    navbarVisibility: true,
+    component: lazy(() => import('#views/Project/Tagging/Assessments/EditAssessment')),
+    componentProps: {
+    },
+    visibility: 'is-authenticated', // TODO handle permission
```
Comment (tnagorra) [inline app/Base/configs/routes.tsx:364]: We need to check for permissions using `checkPermissions`

File: app/views/Project/Tagging/Assessments/ActionCell/index.tsx [hunk 2]
```diff
@@ -84,12 +90,22 @@ function ActionCell(props: Props) {
         }
     }, [deleteAssessment, projectId, assessmentId]);
 
+    const assessmentEditLink = useMemo(() => ({
+        pathname: generatePath(
+            routes.assessmentEdit.path,
+            {
+                projectId,
```
Comment (tnagorra) [inline app/views/Project/Tagging/Assessments/ActionCell/index.tsx]: We do not need to pass projectId in most of the cases. It is automatically injected

File: app/views/Project/Tagging/Assessments/EditAssessment/index.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    const { project } = React.useContext(ProjectContext);
+    const { leadId } = useParams<{ leadId: string }>();
+    const projectId = project ? project.id : undefined;
+    const { className } = props;
+    const aryEndpoint = process.env.REACT_APP_ASSESSMENT_REGISTRY_END;
+    const src = `${aryEndpoint}/projects/${projectId}/leads/${leadId}/ary/edit/`;
+
+    return (
+        <div className={_cs(className, styles.editAssessment)}>
+            {aryEndpoint && (
```
Comment (tnagorra) [inline app/views/Project/Tagging/Assessments/EditAssessment/index.tsx]: Let's add condition on src. and use src definition on 24.

const aryEndpoint = ...;
const src = aryEndpoint ? ... : undefined;

return (
    {src && <iframe />}
)

File: app/views/Project/Tagging/index.tsx [hunk 4]
```diff
@@ -130,6 +130,10 @@ function Tagging(props: Props) {
                     >
                         {subNavbarComponents}
                     </Route>
+                    <Route
+                        exact
+                        path={routes.assessmentEdit.path}
+                    />
```
Comment (tnagorra) [inline app/views/Project/Tagging/index.tsx]: This route is here to load the subNavbarComponents. Please pass them as children

**Review by tnagorra — CHANGES_REQUESTED** [review]

**Review by AdityaKhatri — APPROVED** [review]

**Pull request merged**